scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/policy/ISO8601DateUtil.java

```diff
@@ -74,27 +64,25 @@ private ISO8601DateUtil() {
      * @return ISO8601 complaint string
      */
     public static final String parse(final Date date) {
-        String formatted = df.get().format(date);
-        return formatted.substring(0, 22) + ":" + formatted.substring(22);
+        return new DateTime(date).toString();
     }
 
     /**
-     * Convert ISO8601 complaint string into a java.util.Date instance.
+     * Convert ISO8601 complaint string into a java.util.Date instance. In case
+     * no timezone information is passed, the default local time will be used
      * 
      * @param source
      *            ISO8601 complaint string
      * @return a java.util.Date instance
-     * @throws ParseException
-     *             if the string does not have required length
+     * @throws IllegalArgumentException
+     *             if the string does not have a correct format
      */
-    public static final Date toDate(final String source) throws ParseException {
-        String s = source.replace("Z", "+00:00");
-        try {
-            s = s.substring(0, 22) + s.substring(23);
-        } catch (IndexOutOfBoundsException e) {
-            throw new ParseException("Invalid length", 0);
-        }
-        return df.get().parse(s);
+    public static final Date toDate(final String source) throws IllegalArgumentException {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove the declaration of thrown exception 'java.lang.IllegalArgumentException' which is a runtime exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)